### DISCUSSION TITLE
A way to reset model to initial pose?

### DISCUSSION THREAD

**Discussion by evolvic** [jnsmalm/pixi3d#119] [Q&A] [answered]
Hi,

Firstly can I just say that I absolutely love this framework.
I have been using Pixi heavily for my work since it was first released and the one thing that I always wanted was the ability to have 3d mixed in with the 2d elements and now thanks to this framework, I have!!

I created a Pixi-based layout tool using Pixi and Electron and it has been working great and so I decided to add 3d to the mix and it now makes use of it all.

The issue I am having is that I can't seem to find any easy way to reset a model back to it's initial pose.
Is there something I have missed or is this not possible?

Regards.

**Comment by jnsmalm** [comment]
Hi, thanks!

Do you mean setting it back to initial pose after you have started playing an animation?

**Comment by evolvic** [comment, marked as answer]
Hi,

thanks for the reply buddy and yes, that’s exactly what I am after.
I guess a t-pose animation could be added to each model but this doesn’t solve the problem that I have with current models with animations.

**Reply by jnsmalm** [reply]
You can try if this works for you. If you need the skin at a later time you can just store it somewhere else and add it back and call invalidateShader.

```
model.meshes.forEach(mesh => {
  mesh.skin = undefined
  mesh.material.invalidateShader()
})
```

**Reply by evolvic** [reply]
Ah this is fantastic!
I’m away from my pc until the morning but I will try this first thing tomorrow and let you know how I get on.
Thank you so much for the help buddy, very much appreciated and please keep up the fantastic work.

Regards.

**Reply by evolvic** [reply]
HI,

I just wanted to say that this method works perfectly and I can now play animations as well as restore the model back to it's initial pose whenever I like.

Thank you so much for your help on this, you appear to be incredibly helpful and responsive and it is very much appreciated.

Regards.
